device/transport/mqtt/src/mqtt.ts

```diff
@@ -916,7 +916,7 @@ export class Mqtt extends EventEmitter implements DeviceTransport {
     if (message.contentType) systemProperties['$.ct'] = <string>message.contentType;
     /*Codes_SRS_NODE_DEVICE_MQTT_16_083: [The `sendEvent` method shall serialize the `contentEncoding` property of the message as a key-value pair on the topic with the key `$.ce`.]*/
     if (message.contentEncoding) systemProperties['$.ce'] = <string>message.contentEncoding;
-
+    if (message.interfaceId) systemProperties['$.ifid'] = message.interfaceId;
```

Comment:
>['$.ifid'] [](start = 45, length = 10)

Does this clash with digitaltwins?

Reply:
There is a property sent in digitaltwins registration (which uses the telemetry path) that uses this as a property name.

---
In reply to: [327269387](https://github.com/Azure/azure-iot-sdk-node/pull/633#discussion_r327269387) [](ancestors = 327269387)

Reply:
I was not aware of that, but we are already using that property in other SDKs, so I think it should be fine.
This is only in the context of a D2C message.